website_canonical_url/models/res_config.py

```diff
@@ -8,4 +8,4 @@ class WebsiteConfigSettings(models.TransientModel):
 
     _inherit = 'res.config.settings'
 
-    canonical_domain = fields.Char(related='website_id.canonical_domain')
+    canonical_domain = fields.Char(related='website_id.canonical_domain', required=False)
```

Comment:
why do you remove required? You should set `readonly=False` instead since related fields in v12 are readonly by default.

Reply:
my bad, you are right, i mistaked the word